# BASIC JOURNAL ANALYSIS

The notebook uses _frozen_ static metadata files from the summer of 2022 to analyze the journal-level importance of EarthCube data.


It broadly aims to explore:

1. the proportion of EC papers that have received more citations than the average paper for the journal/year;
2. the cumulative citation count for this group and whether they are higher than the sum of all of these averages;
3. the top 10 papers in terms of % above the average for their journal/year
  

Fixed inputs to this notebook are:

* [../inputs/20220805_ec_journal_titles_plus_citations.xlsx](../inputs/20220805_ec_journal_titles_plus_citations.xlsx): the journal citation exported data from Web of Science
* [../inputs/cr_metadata_20220610012125.json](../inputs/cr_metadata_20220610012125.json): we use a fixed snapshop metadata file extracted from crossref on _June 10, 2022_.  

**NOTE:** We have fixed input snapshot targets because both sources are changing and updating at different times.  Furthermore, WOS data requires subscription, which makes dynamic replication difficult.

In [1]:
import pandas as pd
import json

In [2]:
df = pd.read_excel("../inputs/20220805_ec_journal_titles_plus_citations.xlsx")

In [3]:
df

Unnamed: 0                             doi             journal_title  \
0            15               10.5065/p2jj-9878                        --   
1            79             10.1594/ieda/100709                        --   
2            81          10.5281/zenodo.5496306                        --   
3            93       10.13140/rg.2.1.4908.4561                        --   
4           109  10.6084/m9.figshare.4272164.v1                        --   
..          ...                             ...                       ...   
236         102              10.1111/tgis.12233       Transactions in GIS   
237         219            10.1002/2015wr017342  Water Resources Research   
238          27                  10.22498/pages                       NaN   
239          60   10.17504/protocols.io.fjjbkkn                       NaN   
240         222                  10.1101/647651                       NaN   

                                     publication_title  \
0                                                   --   
1                                                   --   
2                                                   --   
3                                                   --   
4                                                   --   
..                                                 ...   
236  Crowdsensing smart ambient environments and se...   
237  Hydrocomplexity: Addressing water security and...   
238                       Past Global Changes Magazine   
239  ECOGEO 'Omics Training: Introduction to Enviro...   
240  Ecological and genomic attributes of novel bac...   

                                                   url    Year Journal_Avg_cit  
0                    https://doi.org/10.5065/p2jj-9878     NaN              NI  
1    https://doi.org/10.1594/ieda/https://doi.org/1...     NaN              NI  
2               https://doi.org/10.5281/zenodo.5496306     NaN              NI  
3            https://doi.org/10.13140/rg.2.1.4908.4561     NaN              NI  
4       https://doi.org/10.6084/m9.figshare.4272164.v1     NaN              NI  
..                                                 ...     ...             ...  
236                 https://doi.org/10.1111/tgis.12233  2016.0           13.08  
237  https://doi.org/10.https://doi.org/10.2/2015wr...  2015.0           40.97  
238                     https://doi.org/10.22498/pages     NaN              NI  
239      https://doi.org/10.17504/protocols.io.fjjbkkn     NaN              NI  
240     https://doi.org/10.1https://doi.org/10./647651     NaN              NI  

[241 rows x 7 columns]

In [4]:
df['Journal_Avg_cit'] = df['Journal_Avg_cit'].astype(float, errors='ignore')

In [5]:
df

Unnamed: 0                             doi             journal_title  \
0            15               10.5065/p2jj-9878                        --   
1            79             10.1594/ieda/100709                        --   
2            81          10.5281/zenodo.5496306                        --   
3            93       10.13140/rg.2.1.4908.4561                        --   
4           109  10.6084/m9.figshare.4272164.v1                        --   
..          ...                             ...                       ...   
236         102              10.1111/tgis.12233       Transactions in GIS   
237         219            10.1002/2015wr017342  Water Resources Research   
238          27                  10.22498/pages                       NaN   
239          60   10.17504/protocols.io.fjjbkkn                       NaN   
240         222                  10.1101/647651                       NaN   

                                     publication_title  \
0                                                   --   
1                                                   --   
2                                                   --   
3                                                   --   
4                                                   --   
..                                                 ...   
236  Crowdsensing smart ambient environments and se...   
237  Hydrocomplexity: Addressing water security and...   
238                       Past Global Changes Magazine   
239  ECOGEO 'Omics Training: Introduction to Enviro...   
240  Ecological and genomic attributes of novel bac...   

                                                   url    Year Journal_Avg_cit  
0                    https://doi.org/10.5065/p2jj-9878     NaN              NI  
1    https://doi.org/10.1594/ieda/https://doi.org/1...     NaN              NI  
2               https://doi.org/10.5281/zenodo.5496306     NaN              NI  
3            https://doi.org/10.13140/rg.2.1.4908.4561     NaN              NI  
4       https://doi.org/10.6084/m9.figshare.4272164.v1     NaN              NI  
..                                                 ...     ...             ...  
236                 https://doi.org/10.1111/tgis.12233  2016.0           13.08  
237  https://doi.org/10.https://doi.org/10.2/2015wr...  2015.0           40.97  
238                     https://doi.org/10.22498/pages     NaN              NI  
239      https://doi.org/10.17504/protocols.io.fjjbkkn     NaN              NI  
240     https://doi.org/10.1https://doi.org/10./647651     NaN              NI  

[241 rows x 7 columns]

In [6]:
df_subset = df[df['Journal_Avg_cit']!='NI']

In [7]:
df_subset[df_subset['Journal_Avg_cit']>30]

Unnamed: 0                              doi  \
45          134                  10.1145/3129246   
53            7        10.1175/bams-d-14-00164.1   
54           65        10.1175/bams-d-15-00239.1   
70          147        10.1007/s40641-018-0107-0   
109         106        10.1186/s13073-015-0202-y   
110         234                10.1111/gfl.12114   
133          82        10.1109/tgrs.2014.2382566   
178         225                 10.1038/nbt.4306   
179          47        10.1038/s41561-018-0272-8   
206          92     10.1016/j.renene.2017.02.052   
207          44          10.1126/science.aad7048   
208         190  10.1126/science.342.6162.1041-b   
209         180            10.1038/sdata.2017.88   
237         219             10.1002/2015wr017342   

                                         journal_title  \
45                ACM Transactions on Database Systems   
53     Bulletin of the American Meteorological Society   
54     Bulletin of the American Meteorological Society   
70                      Current Climate Change Reports   
109                                    Genome Medicine   
110                                          Geofluids   
133  IEEE Transactions on Geoscience and Remote Sen...   
178                               Nature Biotechnology   
179                                  Nature Geoscience   
206                                   Renewable Energy   
207                                            Science   
208                                            Science   
209                                    Scientific Data   
237                           Water Resources Research   

                                     publication_title  \
45                                         EmptyHeaded   
53   The Earth System Prediction Suite: Toward a Co...   
54   Sharing Experiences and Outlook on Coupling Te...   
70   Rising Oceans Guaranteed: Arctic Land Ice Loss...   
109  Use of semantic workflows to enhance transpare...   
110  DigitalCrust - a 4D data system of material pr...   
133  Regular Shape Similarity Index: A Novel Index ...   
178  Minimum Information about an Uncultivated Viru...   
179  Similarity of fast and slow earthquakes illumi...   
206  Short-term photovoltaic power forecasting usin...   
207          Liberating field science samples and data   
208      Open Data: Crediting a Culture of Cooperation   
209  A global multiproxy database for temperature r...   
237  Hydrocomplexity: Addressing water security and...   

                                                   url    Year Journal_Avg_cit  
45                     https://doi.org/10.1145/3129246  2017.0           35.65  
53           https://doi.org/10.1175/bams-d-14-00164.1  2016.0           30.85  
54           https://doi.org/10.1175/bams-d-15-00239.1  2016.0           30.85  
70   https://doi.org/10.https://doi.org/10.7/s40641...  2018.0              43  
109          https://doi.org/10.1186/s13073-015-0202-y  2015.0           42.88  
110                  https://doi.org/10.1111/gfl.12114  2015.0           30.45  
133  https://doi.org/10.1https://doi.org/10./tgrs.2...  2015.0           43.05  
178   https://doi.org/10.https://doi.org/10.8/nbt.4306  2019.0           67.85  
179  https://doi.org/10.https://doi.org/10.8/s41561...  2019.0            39.7  
206  https://doi.org/10.https://doi.org/10.6/j.rene...  2017.0           32.19  
207            https://doi.org/10.1126/science.aad7048  2016.0              87  
208  https://doi.org/10.1126/science.342.6162.https...  2013.0          126.97  
209  https://doi.org/10.https://doi.org/10.8/sdata....  2017.0           39.61  
237  https://doi.org/10.https://doi.org/10.2/2015wr...  2015.0           40.97

In [8]:
df_json = pd.read_json("../inputs/cr_metadata_20220610012125.json").T

In [9]:
df_json.columns

Index(['indexed', 'reference-count', 'publisher', 'content-domain',
       'short-container-title', 'published-print', 'DOI', 'type', 'created',
       'source', 'is-referenced-by-count', 'title', 'prefix', 'author',
       'member', 'event', 'container-title', 'original-title', 'link',
       'deposited', 'score', 'resource', 'subtitle', 'short-title', 'issued',
       'references-count', 'URL', 'relation', 'published', 'issue', 'license',
       'funder', 'update-policy', 'volume', 'published-online', 'reference',
       'language', 'journal-issue', 'alternative-id', 'archive', 'ISSN',
       'issn-type', 'subject', 'assertion', 'abstract', 'page',
       'published-other', 'accepted', 'publisher-location', 'editor',
       'article-number', 'posted', 'subtype', 'isbn-type', 'ISBN',
       'institution', 'group-title'],
      dtype='object')

In [10]:
df_journals = df.merge(
    df_json[['DOI', 'is-referenced-by-count']].reset_index().drop('index',axis=1).rename(columns={'DOI': 'doi'}),
    on='doi'
)

## Journal Citation Counts vs Actual Citation Counts

Here our interest will turn to how well the actual citation counts match the expected journal citation counts from WOS.

In [11]:
df_journal_cva = df_journals.query('Journal_Avg_cit != "NI"') \
    [["Journal_Avg_cit", "Year", "journal_title", "is-referenced-by-count", 'doi']]

df_journal_cva['Journal_Avg_cit'] = df_journal_cva['Journal_Avg_cit'].astype(int) # note tjos rounds u
df_journal_cva['is-referenced-by-count'] = df_journal_cva['is-referenced-by-count'].astype(int)

In [12]:
df_journal_cva.describe()

Journal_Avg_cit         Year  is-referenced-by-count
count       150.000000   150.000000              150.000000
mean         14.960000  2018.500000               24.173333
std          15.534279     2.071701               47.443042
min           0.000000  2013.000000                0.000000
25%           5.000000  2017.000000                4.000000
50%          12.000000  2019.000000               10.000000
75%          20.750000  2020.000000               21.750000
max         126.000000  2022.000000              344.000000

In [13]:
df_journal_cva[df_journal_cva['is-referenced-by-count']>=df_journal_cva['Journal_Avg_cit']]

Journal_Avg_cit    Year                         journal_title  \
32                35  2017.0  ACM Transactions on Database Systems   
33                 3  2018.0                           AI Magazine   
34                 0  2022.0                  Applied Geochemistry   
38                 9  2020.0                        Biogeosciences   
39                19  2018.0                            BioScience   
..               ...     ...                                   ...   
211                9  2020.0             The Astrophysical Journal   
215                3  2021.0             The Astrophysical Journal   
217                0  2022.0                        The Cryosphere   
222               13  2016.0                   Transactions in GIS   
223               13  2016.0                   Transactions in GIS   

     is-referenced-by-count                              doi  
32                       43                  10.1145/3129246  
33                       13         10.1609/aimag.v39i3.2816  
34                        0  10.1016/j.apgeochem.2022.105273  
38                       14          10.5194/bg-17-2537-2020  
39                       77            10.1093/biosci/biy068  
..                      ...                              ...  
211                      22         10.3847/1538-4357/aba8a6  
215                       4         10.3847/1538-4357/abf2c8  
217                       0          10.5194/tc-16-1431-2022  
222                      31               10.1111/tgis.12232  
223                      16               10.1111/tgis.12233  

[79 rows x 5 columns]

In [14]:
df_journal_cva[['Journal_Avg_cit', 'Year', 'journal_title']].sort_values(by='journal_title')

Journal_Avg_cit    Year                         journal_title
32                35  2017.0  ACM Transactions on Database Systems
33                 3  2018.0                           AI Magazine
34                 0  2022.0                  Applied Geochemistry
35                 2  2021.0    Atmospheric Measurement Techniques
39                19  2018.0                            BioScience
..               ...     ...                                   ...
217                0  2022.0                        The Cryosphere
223               13  2016.0                   Transactions in GIS
222               13  2016.0                   Transactions in GIS
224               40  2015.0              Water Resources Research
163               17  2019.0                                  mBio

[150 rows x 3 columns]

### JOURNAL ANALYSIS

* what is the average journal citation count (over all years)
* grouping by journal what are the paper averages

In [15]:
journals = df_journal_cva['journal_title'].unique()

In [16]:
df_journal_cva[['Journal_Avg_cit', 'journal_title']].groupby('journal_title').mean()


Journal_Avg_cit
journal_title                                               
ACM Transactions on Database Systems                    35.0
AI Magazine                                              3.0
Applied Geochemistry                                     0.0
Atmospheric Measurement Techniques                       2.0
BioScience                                              19.0
...                                                      ...
The Astrophysical Journal Supplement Series             17.0
The Cryosphere                                           0.0
Transactions in GIS                                     13.0
Water Resources Research                                40.0
mBio                                                    17.0

[89 rows x 1 columns]

In [17]:
df_journal_cva[['Journal_Avg_cit', 'journal_title']].groupby('journal_title').count().sort_values(by='Journal_Avg_cit', ascending=False)

Journal_Avg_cit
journal_title                                                  
The Astrophysical Journal                                    12
Environmental Modelling &amp; Software                        8
ISPRS International Journal of Geo-Information                6
Computers &amp; Geosciences                                   5
Journal of Geophysical Research: Space Physics                5
...                                                         ...
International Journal of Digital Earth                        1
International Journal of Remote Sensing                       1
International Journal of Semantic Computing                   1
Journal of Atmospheric and Oceanic Technology                 1
mBio                                                          1

[89 rows x 1 columns]

In [18]:
df_journal_density_counts = \
    df_journal_cva[['Journal_Avg_cit', 'journal_title']]\
    .groupby('journal_title').count()\
    .sort_values(by='Journal_Avg_cit', ascending=False)
df_journal_density_counts = df_journal_density_counts.rename(columns={'Journal_Avg_cit': 'paper_counts'})

df_journal_density_counts

paper_counts
journal_title                                               
The Astrophysical Journal                                 12
Environmental Modelling &amp; Software                     8
ISPRS International Journal of Geo-Information             6
Computers &amp; Geosciences                                5
Journal of Geophysical Research: Space Physics             5
...                                                      ...
International Journal of Digital Earth                     1
International Journal of Remote Sensing                    1
International Journal of Semantic Computing                1
Journal of Atmospheric and Oceanic Technology              1
mBio                                                       1

[89 rows x 1 columns]

In [19]:
df_journal_density_means = \
    df_journal_cva[['Journal_Avg_cit', 'journal_title']].groupby('journal_title').mean().sort_values(by='Journal_Avg_cit', ascending=False)

df_journal_density_means = df_journal_density_means.rename(columns={'Journal_Avg_cit': 'mean_cites'})

df_journal_density_means

mean_cites
journal_title                                                 
Science                                                  106.5
Nature Biotechnology                                      67.0
Current Climate Change Reports                            43.0
IEEE Transactions on Geoscience and Remote Sensing        43.0
Genome Medicine                                           42.0
...                                                        ...
Engaging Science, Technology, and Society                  0.0
Remote Sensing in Ecology and Conservation                 0.0
The Cryosphere                                             0.0
Data in Brief                                              0.0
Applied Geochemistry                                       0.0

[89 rows x 1 columns]

For all journals with 2 or more papers, how did they do:

In [20]:
df_journal_cva.columns

Index(['Journal_Avg_cit', 'Year', 'journal_title', 'is-referenced-by-count',
       'doi'],
      dtype='object')

### JOURNALS WITH MORE THAN 1 PUBLICATION

In [21]:
journal_list = df_journal_density_counts[df_journal_density_counts['paper_counts']>1].index.to_list()

df_journal_ec_mean_analysis = \
    df_journal_cva[df_journal_cva['journal_title'].isin(journal_list)]\
        .groupby('journal_title').mean() \
        .drop(columns=['Year']) \
        .merge(df_journal_density_counts, left_index=True, right_index=True) \
        .sort_values('paper_counts', ascending=False) \
        .rename(columns={'Journal_Avg_cit': 'journal_mean_cites', 'is-referenced-by-count': 'ec_mean_cites'})

df_journal_ec_mean_analysis.describe()

journal_mean_cites  ec_mean_cites  paper_counts
count           27.000000      27.000000     27.000000
mean            15.389198      22.391975      3.259259
std             19.555807      33.431244      2.297031
min              1.000000       0.500000      2.000000
25%              7.450000       8.600000      2.000000
50%             12.000000      11.500000      2.000000
75%             17.062500      23.875000      3.000000
max            106.500000     175.000000     12.000000

In [22]:
df_journal_ec_mean_analysis.paper_counts.sum()

88

* **88 papers (58.67%) with 2 or more** papers in a single journal (27 distinct journals)

In [23]:
df_journal_ec_mean_analysis

journal_mean_cites  \
journal_title                                                            
The Astrophysical Journal                                    10.833333   
Environmental Modelling &amp; Software                       17.125000   
ISPRS International Journal of Geo-Information                9.166667   
Journal of Geophysical Research: Space Physics                7.400000   
Computers &amp; Geosciences                                   8.400000   
JAWRA Journal of the American Water Resources A...           13.750000   
Bulletin of the American Meteorological Society              28.333333   
Concurrency and Computation: Practice and Exper...            1.000000   
Earth and Space Science                                      15.333333   
Journal of Proteome Research                                  7.666667   
Future Generation Computer Systems                           17.666667   
Journal of Geophysical Research: Solid Earth                  5.333333   
Semantic Web                                                 12.500000   
Science                                                     106.500000   
Remote Sensing                                               13.500000   
Journal of Hydroinformatics                                  12.000000   
Geophysical Research Letters                                 17.000000   
GigaScience                                                  20.000000   
Geosphere                                                     5.000000   
Geophysical Journal International                             6.000000   
Geology                                                      22.000000   
Frontiers in Microbiology                                     8.500000   
Frontiers in Astronomy and Space Sciences                     2.000000   
Earth Science Informatics                                     7.500000   
Computing in Science &amp; Engineering                        1.000000   
Computers, Environment and Urban Systems                     27.000000   
Transactions in GIS                                          13.000000   

                                                    ec_mean_cites  \
journal_title                                                       
The Astrophysical Journal                                9.166667   
Environmental Modelling &amp; Software                  11.500000   
ISPRS International Journal of Geo-Information          10.500000   
Journal of Geophysical Research: Space Physics          10.800000   
Computers &amp; Geosciences                              8.200000   
JAWRA Journal of the American Water Resources A...      24.250000   
Bulletin of the American Meteorological Society         34.666667   
Concurrency and Computation: Practice and Exper...       2.000000   
Earth and Space Science                                 34.000000   
Journal of Proteome Research                            17.666667   
Future Generation Computer Systems                       7.666667   
Journal of Geophysical Research: Solid Earth             4.666667   
Semantic Web                                            13.000000   
Science                                                 30.000000   
Remote Sensing                                           4.000000   
Journal of Hydroinformatics                             10.000000   
Geophysical Research Letters                           175.000000   
GigaScience                                             17.500000   
Geosphere                                                6.000000   
Geophysical Journal International                       10.500000   
Geology                                                 41.500000   
Frontiers in Microbiology                               11.500000   
Frontiers in Astronomy and Space Sciences                0.500000   
Earth Science Informatics                               16.500000   
Computing in Science &amp; Engineering                   9.000000   
Computers, Environment and Urban Systems  

In [24]:
df_journal_ec_mean_analysis[ 
    df_journal_ec_mean_analysis.ec_mean_cites>=df_journal_ec_mean_analysis.journal_mean_cites
].paper_counts.sum()

47

* **47 of 88 papers (53.4%)** with a mean_cite greater than or equal to journal_mean_cite (for all years, for journals where there were more than 1 publication)

### JOURNALS WITH ONLY 1 PUBLICATION

In [25]:
journal_list = df_journal_density_counts[df_journal_density_counts['paper_counts']<2].index.to_list()

df_journal_ec_mean_analysis = \
    df_journal_cva[df_journal_cva['journal_title'].isin(journal_list)]\
        .groupby('journal_title').mean() \
        .drop(columns=['Year']) \
        .merge(df_journal_density_counts, left_index=True, right_index=True) \
        .sort_values('paper_counts', ascending=False) \
        .rename(columns={'Journal_Avg_cit': 'journal_mean_cites', 'is-referenced-by-count': 'ec_mean_cites'})

df_journal_ec_mean_analysis.describe()

journal_mean_cites  ec_mean_cites  paper_counts
count           62.000000      62.000000          62.0
mean            16.370968      32.387097           1.0
std             13.698431      56.452027           0.0
min              0.000000       0.000000           1.0
25%              6.250000       3.250000           1.0
50%             12.500000      13.500000           1.0
75%             23.000000      38.500000           1.0
max             67.000000     344.000000           1.0

* **62 of 150 papers** with only 1 publication in a journal
* **journal means = 16.37, earthcube paper means = 32.39**
* stdevs are very different (a lot more variance)

In [26]:
df_journal_ec_mean_analysis

journal_mean_cites  \
journal_title                                                            
ACM Transactions on Database Systems                              35.0   
PLoS ONE                                                          26.0   
Journal of Climate                                                14.0   
Journal of Environmental Quality                                  21.0   
Journal of Sedimentary Research                                   10.0   
...                                                                ...   
Hydrological Processes                                            21.0   
IEEE Systems Journal                                              25.0   
IEEE Transactions on Geoscience and Remote Sensing                43.0   
IEEE Transactions on Parallel and Distributed S...                24.0   
mBio                                                              17.0   

                                                    ec_mean_cites  \
journal_title                                                       
ACM Transactions on Database Systems                         43.0   
PLoS ONE                                                     54.0   
Journal of Climate                                           12.0   
Journal of Environmental Quality                             73.0   
Journal of Sedimentary Research                               4.0   
...                                                           ...   
Hydrological Processes                                       16.0   
IEEE Systems Journal                                          5.0   
IEEE Transactions on Geoscience and Remote Sensing           23.0   
IEEE Transactions on Parallel and Distributed S...            2.0   
mBio                                                         55.0   

                                                    paper_counts  
journal_title                                                     
ACM Transactions on Database Systems                           1  
PLoS ONE                                                       1  
Journal of Climate                                             1  
Journal of Environmental Quality                               1  
Journal of Sedimentary Research                                1  
...                                                          ...  
Hydrological Processes                                         1  
IEEE Systems Journal                                           1  
IEEE Transactions on Geoscience and Remote Sensing             1  
IEEE Transactions on Parallel and Distributed S...             1  
mBio                                                           1  

[62 rows x 3 columns]

In [27]:
df_journal_ec_mean_analysis[ 
    df_journal_ec_mean_analysis.ec_mean_cites>=df_journal_ec_mean_analysis.journal_mean_cites
].paper_counts.sum()

32

In [28]:
df_journal_ec_mean_analysis[ 
    df_journal_ec_mean_analysis.ec_mean_cites>df_journal_ec_mean_analysis.journal_mean_cites
].paper_counts.sum()

26

* **26 of 62 (41.9%) EC papers produce more than mean citations** than the journal means
* **32 of 62 (51.6%) EC papers produce at least the mean citations** from the journal means

* the top 10 papers in terms of % above the average for their journal/year

In [29]:
df_journal_cva

Journal_Avg_cit    Year                                journal_title  \
32                35  2017.0         ACM Transactions on Database Systems   
33                 3  2018.0                                  AI Magazine   
34                 0  2022.0                         Applied Geochemistry   
35                 2  2021.0           Atmospheric Measurement Techniques   
38                 9  2020.0                               Biogeosciences   
..               ...     ...                                          ...   
216               17  2020.0  The Astrophysical Journal Supplement Series   
217                0  2022.0                               The Cryosphere   
222               13  2016.0                          Transactions in GIS   
223               13  2016.0                          Transactions in GIS   
224               40  2015.0                     Water Resources Research   

     is-referenced-by-count                              doi  
32                       43                  10.1145/3129246  
33                       13         10.1609/aimag.v39i3.2816  
34                        0  10.1016/j.apgeochem.2022.105273  
35                        0         10.5194/amt-14-6917-2021  
38                       14          10.5194/bg-17-2537-2020  
..                      ...                              ...  
216                       2         10.3847/1538-4365/aba4aa  
217                       0          10.5194/tc-16-1431-2022  
222                      31               10.1111/tgis.12232  
223                      16               10.1111/tgis.12233  
224                      34             10.1002/2015wr017342  

[150 rows x 5 columns]

In [30]:
df_journal_cva_reindexed = df_journal_cva.rename(columns={
    'Journal_Avg_cit': 'journal_mean_cites',
    'Year': 'year',
    'is-referenced-by-count': 'ec_cites'
    }).reset_index().drop('index', axis=1)
df_journal_cva_reindexed

journal_mean_cites    year                                journal_title  \
0                    35  2017.0         ACM Transactions on Database Systems   
1                     3  2018.0                                  AI Magazine   
2                     0  2022.0                         Applied Geochemistry   
3                     2  2021.0           Atmospheric Measurement Techniques   
4                     9  2020.0                               Biogeosciences   
..                  ...     ...                                          ...   
145                  17  2020.0  The Astrophysical Journal Supplement Series   
146                   0  2022.0                               The Cryosphere   
147                  13  2016.0                          Transactions in GIS   
148                  13  2016.0                          Transactions in GIS   
149                  40  2015.0                     Water Resources Research   

     ec_cites                              doi  
0          43                  10.1145/3129246  
1          13         10.1609/aimag.v39i3.2816  
2           0  10.1016/j.apgeochem.2022.105273  
3           0         10.5194/amt-14-6917-2021  
4          14          10.5194/bg-17-2537-2020  
..        ...                              ...  
145         2         10.3847/1538-4365/aba4aa  
146         0          10.5194/tc-16-1431-2022  
147        31               10.1111/tgis.12232  
148        16               10.1111/tgis.12233  
149        34             10.1002/2015wr017342  

[150 rows x 5 columns]

In [31]:
# filter on items that ec beats (or is equal to) the journal mean
df_journal_cva_reindexed[
    df_journal_cva_reindexed.ec_cites>=df_journal_cva_reindexed.journal_mean_cites
]

# cast year as int
df_journal_cva_reindexed.year = df_journal_cva_reindexed.year.astype(int)

# norm the pct diff (i.e. 200% = "100% more")
df_journal_cva_reindexed['ec_pct_diff'] = \
    100*((df_journal_cva_reindexed.ec_cites / df_journal_cva_reindexed.journal_mean_cites) - 1)

* top 20 Earthcube papers by pct diff from journal mean

In [90]:
df_tmp = df_journal_cva_reindexed.sort_values(by='ec_pct_diff', ascending=False) \
    .query("journal_mean_cites > 0")[:20]
# df_tmp.doi = df_tmp.doi.apply(lambda d: f"[{d}](https://doi.org/{d})")
df_tmp.columns = ['Journal Mean Cites', 'Publication Year', 'Journal Title', 'EC Cites', 'EC Publication DOI', '% diff mean cites']
df_tmp.to_csv("../outputs/ec_journals_top_pct_delta.csv")

In [91]:
df_tmp

Journal Mean Cites  Publication Year  \
74                   17              2017   
61                   25              2017   
117                  10              2018   
18                    1              2021   
10                    4              2019   
19                    1              2021   
68                    4              2021   
75                    8              2019   
28                   15              2016   
85                   13              2017   
106                   5              2017   
25                    8              2020   
1                     3              2018   
125                  39              2017   
5                    19              2018   
60                    1              2021   
17                   27              2017   
20                    1              2021   
122                  32              2017   
100                  12              2019   

                                         Journal Title  EC Cites  \
74              International Journal of Digital Earth       344   
61                        Geophysical Research Letters       336   
117                                Quaternary Research       120   
18              Computing in Science &amp; Engineering        10   
10                           Communications of the ACM        39   
19              Computing in Science &amp; Engineering         8   
68                                        GSA Bulletin        30   
75             International Journal of Remote Sensing        49   
28                             Earth and Space Science        91   
85   JAWRA Journal of the American Water Resources ...        72   
106                  Journal of Visualized Experiments        25   
25                                   Data Intelligence        40   
1                                          AI Magazine        13   
125                                    Scientific Data       167   
5                                           BioScience        77   
60                   Geophysical Journal International         4   
17            Computers, Environment and Urban Systems       108   
20   Concurrency and Computation: Practice and Expe...         4   
122                                   Renewable Energy       127   
100                       Journal of Proteome Research        44   

                      EC Publication DOI  % diff mean cites  
74         10.1080/17538947.2016.1239771        1923.529412  
61                  10.1002/2017gl074954        1244.000000  
117                 10.1017/qua.2017.105        1100.000000  
18             10.1109/mcse.2021.3059437         900.000000  
10                       10.1145/3192335         875.000000  
19             10.1109/mcse.2021.3059263         700.000000  
68                      10.1130/b35560.1         650.000000  
75         10.1080/01431161.2018.1516313         512.500000  
28                  10.1002/2015ea000136         506.666667  
85               10.1111/1752-1688.12474         453.846154  
106                        10.3791/54660         400.000000  
25                  10.1162/dint_a_00033         400.000000  
1               10.1609/aimag.v39i3.2816         333.333333  
125                10.1038/sdata.2017.88         328.205128  
5                  10.1093/biosci/biy068         305.263158  
60                   10.1093/gji/ggab238         300.000000  
17   10.1016/j.compenvurbsys.2016.10.010         300.000000  
20                      10.1002/cpe.6099         300.000000  
122         10.1016/j.renene.2017.02.052         296.875000  
100        10.1021/acs.jproteome.8b00761         266.666667

In [92]:
df_citations = pd.read_csv("../outputs/full_nsf_doi_project_summary.tsv", sep='\t')[['doi', 'ams_bib']]

In [118]:

print("""
|Citation| EC Cites | Journal Mean Cites | % diff mean cites|
|---:|:--:|:--:|:--:|""")
for i, r in df_tmp[:10].iterrows():
    doi = df_citations.query(f"doi == '{r['EC Publication DOI']}'").drop_duplicates()['ams_bib'].unique()[0]
    print(f"{doi} | {r['EC Cites']} | {r['Journal Mean Cites']} | {r['% diff mean cites']:.2f}% |")


|Citation| EC Cites | Journal Mean Cites | % diff mean cites|
|---:|:--:|:--:|:--:|
Yang, C., Q. Huang, Z. Li, K. Liu, and F. Hu, 2016: Big Data and cloud computing: innovation opportunities and challenges. International Journal of Digital Earth, 10, 13–53, https://doi.org/10.1080/17538947.2016.1239771. | 344 | 17 | 1923.53% |
Morlighem, M., and Coauthors, 2017: BedMachine v3: Complete Bed Topography and Ocean Bathymetry Mapping of Greenland From Multibeam Echo Sounding Combined With Mass Conservation. Geophysical Research Letters, 44, https://doi.org/10.1002/2017gl074954. | 336 | 25 | 1244.00% |
Williams, J. W., and Coauthors, 2018: The Neotoma Paleoecology Database, a multiproxy, international, community-curated data resource. Quaternary Research, 89, 156–177, https://doi.org/10.1017/qua.2017.105. | 120 | 10 | 1100.00% |
Abernathey, R. P., and Coauthors, 2021: Cloud-Native Repositories for Big Scientific Data. Computing in Science &amp; Engineering, 23, 26–35, https://doi.org/10.110

|Citation| EC Cites | Journal Mean Cites | % diff mean cites|
|---:|:--:|:--:|:--:|
Yang, C., Q. Huang, Z. Li, K. Liu, and F. Hu, 2016: Big Data and cloud computing: innovation opportunities and challenges. International Journal of Digital Earth, 10, 13–53, https://doi.org/10.1080/17538947.2016.1239771. | 344 | 17 | 1923.53% |
Morlighem, M., and Coauthors, 2017: BedMachine v3: Complete Bed Topography and Ocean Bathymetry Mapping of Greenland From Multibeam Echo Sounding Combined With Mass Conservation. Geophysical Research Letters, 44, https://doi.org/10.1002/2017gl074954. | 336 | 25 | 1244.00% |
Williams, J. W., and Coauthors, 2018: The Neotoma Paleoecology Database, a multiproxy, international, community-curated data resource. Quaternary Research, 89, 156–177, https://doi.org/10.1017/qua.2017.105. | 120 | 10 | 1100.00% |
Abernathey, R. P., and Coauthors, 2021: Cloud-Native Repositories for Big Scientific Data. Computing in Science &amp; Engineering, 23, 26–35, https://doi.org/10.1109/mcse.2021.3059437. | 10 | 1 | 900.00% |
Gil, Y., and Coauthors, 2018: Intelligent systems for geosciences. Communications of the ACM, 62, 76–84, https://doi.org/10.1145/3192335. | 39 | 4 | 875.00% |
Granger, B. E., and F. Perez, 2021: Jupyter: Thinking and Storytelling With Code and Data. Computing in Science &amp; Engineering, 23, 7–14, https://doi.org/10.1109/mcse.2021.3059263. | 8 | 1 | 700.00% |
Schaen, A. J., and Coauthors, 2020: Interpreting and reporting 40Ar/39Ar geochronologic data. GSA Bulletin, 133, 461–487, https://doi.org/10.1130/b35560.1. | 30 | 4 | 650.00% |
Sun, Z., L. Di, and H. Fang, 2018: Using long short-term memory recurrent neural network in land cover classification on Landsat and Cropland data layer time series. International Journal of Remote Sensing, 40, 593–614, https://doi.org/10.1080/01431161.2018.1516313. | 49 | 8 | 512.50% |
Gil, Y., and Coauthors, 2016: Toward the Geoscience Paper of the Future: Best practices for documenting and sharing research from data to software to provenance. Earth and Space Science, 3, 388–415, https://doi.org/10.1002/2015ea000136. | 91 | 15 | 506.67% |
Maidment, D. R., 2016: Conceptual Framework for the National Flood Interoperability Experiment. JAWRA Journal of the American Water Resources Association, 53, 245–257, https://doi.org/10.1111/1752-1688.12474. | 72 | 13 | 453.85% |

**TAKEAWAYS**

* journal average citation: 14.96
* average citation of EC papers: 24.17 *from WOS metadata citation counts
* EC papers beat average 79 times (52.7% of the time)


In [35]:
df_journal_cva[df_journal_cva['Journal_Avg_cit']>20]

Journal_Avg_cit    Year  \
32                35  2017.0   
40                30  2016.0   
41                30  2016.0   
42                25  2018.0   
50                27  2017.0   
51                27  2017.0   
57                43  2018.0   
61                26  2017.0   
70                28  2017.0   
72                28  2017.0   
73                28  2017.0   
84                23  2017.0   
93                24  2017.0   
94                24  2017.0   
96                42  2015.0   
97                30  2015.0   
101               22  2017.0   
102               22  2017.0   
103               26  2015.0   
106               25  2017.0   
108               21  2016.0   
118               21  2016.0   
119               25  2015.0   
120               43  2015.0   
121               24  2016.0   
141               21  2015.0   
165               67  2019.0   
166               39  2019.0   
179               26  2014.0   
189               25  2017.0   
192               23  2020.0   
193               32  2017.0   
194               87  2016.0   
195              126  2013.0   
196               39  2017.0   
201               23  2015.0   
214               21  2017.0   
224               40  2015.0   

                                         journal_title  \
32                ACM Transactions on Database Systems   
40     Bulletin of the American Meteorological Society   
41     Bulletin of the American Meteorological Society   
42     Bulletin of the American Meteorological Society   
50            Computers, Environment and Urban Systems   
51            Computers, Environment and Urban Systems   
57                      Current Climate Change Reports   
61                 Earth and Planetary Science Letters   
70              Environmental Modelling &amp; Software   
72              Environmental Modelling &amp; Software   
73              Environmental Modelling &amp; Software   
84                           FEMS Microbiology Ecology   
93                  Future Generation Computer Systems   
94                  Future Generation Computer Systems   
96                                     Genome Medicine   
97                                           Geofluids   
101                                            Geology   
102                                            Geology   
103                                      Geomorphology   
106                       Geophysical Research Letters   
108                            Geoscience Data Journal   
118                             Hydrological Processes   
119                               IEEE Systems Journal   
120  IEEE Transactions on Geoscience and Remote Sen...   
121  IEEE Transactions on Parallel and Distributed ...   
141                   Journal of Environmental Quality   
165                               Nature Biotechnology   
166                                  Nature Geoscience   
179                                           PLoS ONE   
189                                     Remote Sensing   
192                      Remote Sensing of Environment   
193                                   Renewable Energy   
194                                            Science   
195                                            Science   
196                                    Scientific Data   
201                                     Sustainability   
214                          The Astrophysical Journal   
224                           Water Resources Research   

     is-referenced-by-count                                  doi  
32                       43                      10.1145/3129246  
40                       33            10.1175/bams-d-14-00164.1  
41                        6            10.1175/bams-d-15-00239.1  
42                       65             10.1175/bams-d-16-0215.1  
50                       14  10.1016/j.compenvurbsys.2016.11.007  
51                      108  10.1016/j.compenvurbsys.2016.10.010  
57                      